frontend/src/metabase/search/containers/SearchApp.jsx

```diff
@@ -25,172 +30,90 @@ import {
-                    <div className="flex justify-end my2">
-                      <PaginationControls
-                        showTotal
-                        pageSize={PAGE_SIZE}
-                        page={page}
-                        itemsLength={list.length}
-                        total={metadata.total}
-                        onNextPage={handleNextPage}
-                        onPreviousPage={handlePreviousPage}
-                      />
-                    </div>
-                  </Fragment>
-                </SearchMain>
+      <Search.ListLoader query={query} wrapped>
+        {({ list, metadata }) =>
+          list && list.length > 0 ? (
+            <SearchBody>
+              <SearchMain>
+                <SearchResultSection items={list} />
+                <PaginationControls
+                  showTotal
+                  pageSize={PAGE_SIZE}
+                  page={page}
+                  itemsLength={list.length}
+                  total={metadata.total}
+                  onNextPage={handleNextPage}
+                  onPreviousPage={handlePreviousPage}
+                />
+              </SearchMain>
+              {searchFilters?.type || metadata.available_models ? (
                 <SearchControls>
-                  {filters.length > 0 ? (
-                    <Link
-                      className={cx("flex align-center mb3", {
-                        "text-brand": filter == null,
-                        "text-inherit": filter != null,
-                      })}
-                      onClick={() => handleFilterChange(null)}
-                      to={{
-                        pathname: location.pathname,
-                        query: { ...location.query, type: undefined },
-                      }}
-                    >
-                      <Icon name="search" className="mr1" />
-                      <h4>{t`All results`}</h4>
-                    </Link>
-                  ) : null}
-                  {filters.map(f => {
-                    const isActive = filter === f.filter;
-
-                    return (
-                      <Link
-                        key={f.filter}
-                        className={cx("mb3 flex align-center", {
-                          "text-brand": isActive,
-                          "text-medium": !isActive,
-                        })}
-                        onClick={() => handleFilterChange(f)}
-                        to={{
-                          pathname: location.pathname,
-                          query: { ...location.query, type: f.filter },
-                        }}
-                      >
-                        <Icon className="mr1" name={f.icon} size={16} />
-                        <h4>{f.name}</h4>
-                      </Link>
-                    );
-                  })}
+                  <TypeSearchSidebar
+                    availableModels={metadata.available_models.filter(
```

Comment:
The condition before rendering this component looks out of place with this line `searchFilters?.type || metadata.available_models`

That condition checks if `metadata.available_models` is not null. However, the condition could be true if only `searchFilters.type` is not null, but `metadata.available_models` is null. This line use `metadata.available_models` as if it's always an array which doesn't seem to go along with the condition before rendering this component.

Can you make sure everything here is good? I think something probably needs to be changed, I'm not sure which one.

Reply:
I see what you're saying - the goal of the check was to make sure that we were getting _either_ the full set of `available_models` _or_ getting the intersection of `available_models` and `searchFilters.type` ,so we only show types that were found during the search, which is not necessarily equal to the user's selected type filters. However, if the user doesn't specify a `type` filter, we just want to use all `available_models`. 

I see now that if we just keep this `searchFilters?.type` boolean then we're good, since if there's no `available_models` that means the user doesn't have any data in their system and we should have a different error entirely like a "Add data" message

Reply:
Thanks for explaining the condition. I mean I can understand what it does, but yeah if we remove `|| metadata.available_models`, it will be more obvious.